internal/pkg/propagation/propagation.go

```diff
@@ -417,7 +417,7 @@ func hasTaintableType(n ssa.Node) bool {
 	if v, ok := n.(ssa.Value); ok {
 		switch t := v.Type().(type) {
 		case *types.Basic:
-			return t.Info() != types.IsBoolean
+			return (t.Info() & types.IsString) != 0
```

Comment:
Not to gild a rose here, but should we check non-basic types?  While we won't taint, e.g., `float32`, we would still taint a `[]float32`, correct?

I don't remember where we left type inference, but that could be extended to include exemption for all non-interface-or-string-like-reachable structs, e.g. `type Coordinate struct { x, y int }` and `map[int]int` would be implicitly untaintable, being composed only of untaintable data.  That could be work for future PRs, though.  Just wondering your thoughts and if it's worth opening an `Issue` for it.

Reply:
You are correct. I'm personally ok with waiting for e.g. `[]float32` to cause problems, like we did for e.g. `int` in the case of this PR.

I think what you bring up concerning type inference is interesting. We would need to be careful to not exclude e.g. `[]byte`, as that it basically the same thing as a `string`.

IMO this is definitely work for future PRs. I'm not sure it's worth its own issue, but it should at the very least be a comment on the existing source inference issue(s).

Reply:
> IMO this is definitely work for future PRs. [...]

Sounds good to me.

And yeah, there's all the special handling around the general castability between `string`, `[]byte` (alias for `[]uint8`), and `[]rune` (alias for `[]int32`).  Because why should anything be clean cut and simple?